release/rancher/rancher.go

```diff
@@ -274,7 +278,68 @@ func createPRFromRancher(ctx context.Context, rancherBaseBranch, title, branchNa
 		Head:                github.String(forkOwner + ":" + branchName),
 		MaintainerCanModify: github.Bool(true),
 	}
-	_, _, err = ghClient.PullRequests.Create(ctx, org, repo, pull)
+	_, _, err = ghClient.PullRequests.Create(ctx, org, rancherRepo, pull)
 
 	return err
 }
+
+func CheckRancherFinalRCDeps(org, repo, commitHash, releaseTitle, files string) error {
+	var (
+		devDependencyPattern = regexp.MustCompile(`dev-v[0-9]+\.[0-9]+`)
+		rcTagPattern         = regexp.MustCompile(`-rc[0-9]+`)
+		matchCommitMessage   bool
+		existsReleaseTitle   bool
+		badFiles             bool
+	)
+
+	httpClient := ecmHTTP.NewClient(time.Second * 15)
+
+	if repo == "" {
+		repo = rancherRepo
+	}
+	if org == "" {
+		org = rancherOrg
+	}
+	if commitHash != "" {
+		commitData, err := repository.CommitInfo(org, repo, commitHash, &httpClient)
+		if err != nil {
+			return err
+		}
+		matchCommitMessage = strings.Contains(commitData.Message, partialFinalRCCommitMessage)
+
+	}
+	if releaseTitle != "" {
+		innerExistsReleaseTitle, err := regexp.MatchString(releaseTitleRegex, releaseTitle)
+		if err != nil {
+			return err
+		}
+		existsReleaseTitle = innerExistsReleaseTitle
+	}
+
+	if matchCommitMessage || existsReleaseTitle {
+		for _, filePath := range strings.Split(files, ",") {
+			content, err := repository.ContentByFileNameAndCommit(org, repo, commitHash, filePath, &httpClient)
+			if err != nil {
+				return err
+			}
+
+			if devDependencyPattern.Match(content) {
```

Comment:
Let's record the name of the file and possibly the the line number and return that as output to the user.